core/src/stage.cpp

```diff
@@ -283,6 +283,13 @@ void Stage::setName(const std::string& name) {
 	pimpl_->name_ = name;
 }
 
+uint32_t Stage::introspectionId() const {
+	if (!pimpl_->introspection_)
+		throw std::runtime_error(
+		    "Introspection not available. Initialize the full Task before calling introspectionId()");
```

Comment:
It's not about initializing the task, but about enabling introspection.
```suggestion
		throw std::runtime_error("Introspection wasn't enabled.");
```

Reply:
No, it's actually about initialization because the `introspection_` instance is only passed to the stages during init and the stages register with the instance only once the first description is published via `fillTaskDescription`.

Reply:
You are right. I updated the error text proposal.

Reply:
I pushed a rephrased version.